src/Avalonia.Controls/Primitives/TemplatedControl.cs

```diff
@@ -333,6 +335,8 @@ protected virtual void OnApplyTemplate(TemplateAppliedEventArgs e)
         /// </summary>
         /// <param name="e">The event args.</param>
         [Obsolete("Use OnApplyTemplate")]
+        [System.ComponentModel.Browsable(false)]
```

Comment:
I don't know what your policy is on making an api obsolete.
I use this policy.

- First major version -> marked as obsolete the api
- Second major version -> I hide the API to prevent new developers from using the deprecated API
- Third major version -> removal of the obsolete API